titration/utils/ui_state/update_settings/update_settings.py

```diff
@@ -1,67 +1,110 @@
+"""
+The file to for the UpdateSetting class
+"""
 from titration.utils.ui_state.ui_state import UIState
 from titration.utils import lcd_interface
 from titration.utils.ui_state.user_value.user_value import UserValue
 
 
 class UpdateSettings(UIState):
-    def __init__(self, titrator, state):
-        super().__init__(titrator, state)
+    """
+    This is a class for the UpdateSettings state of the titrator
+
+    Attributes:
+        titrator (Titrator object): the titrator is used to move through the state machine
+        previous_state (UIState object): the previous_state is used to return the last visited state
+        substate (int): the substate is used to keep track of substate of the UIState
+        values (dict): values is a dictionary to hold the volume in the pump
+    """
+
+    def __init__(self, titrator, previous_state):
+        """
+        The constructor for the UpdateSettings class
+
+        Parameters:
+            titrator (Titrator object): the titrator is used to move through the state machine
+            previous_state (UIState object): the previous_state is used to return the last visited state
+        """
+        super().__init__(titrator, previous_state)
         self.values = {"vol_in_pump": 0}
 
-    def name(self):
-        return "UpdateSettings"
+    def handle_key(self, key):
+        """
+        The function to respond to a keypad input:
+            Substate 1:
+                Y -> Reset calibration settings to default
+                N -> Do not reset calibration settings to default
+            Substate 2:
+                Any -> To continue
+            Substate 3:
+                Y -> Set volume in pump
+                N -> Do not set volume in pump
+            Substate 4:
+                Any -> Enter user value state to set volume in pump
+            Substate 5:
+                Any -> Return to previous state
 
-    def handleKey(self, key):
-        if self.subState == 1:
-            if key != "n" and key != "N":
-                self.subState += 1
-            else:
-                self.subState += 2
+        Parameters:
+            key (char): the keypad input is used to move through the substates
+        """
+        if self.substate == 1:
+            if key == "Y" or key == "y":
```

Comment:
Does this switch the menu steps?

Reply:
This does not switch the menu steps. The menu asks for the user to enter "y" before moving to the next state. Prior to my change, a user could have pressed any button to continue. This makes the handle_key function consistent with what the menu is displaying